python/mxnet/gluon/model_zoo/vision/mobilenet.py

```diff
@@ -74,13 +123,68 @@ def hybrid_forward(self, F, x):
         x = self.output(x)
         return x
 
+
+class MobileNetV2(nn.HybridBlock):
+    r"""MobileNetV2 model from the
+    `"Inverted Residuals and Linear Bottlenecks:
+      Mobile Networks for Classification, Detection and Segmentation"
+    <https://arxiv.org/abs/1801.04381>`_ paper.
+
+    Parameters
+    ----------
+    multiplier : float, default 1.0
+        The width multiplier for controling the model size. The actual number of channels
+        is equal to the original channel size multiplied by this multiplier.
+    classes : int, default 1000
+        Number of classes for the output layer.
+    """
+
+    def __init__(self, multiplier=1.0, classes=1000, **kwargs):
+        super(MobileNetV2, self).__init__(**kwargs)
+        with self.name_scope():
+            self.features = nn.HybridSequential(prefix='features_')
+            with self.features.name_scope():
+                _add_conv(self.features, int(32 * multiplier), kernel=3, stride=2, pad=1)
+
+                in_channels_group = [int(x * multiplier) for x in [32] + [16] + [24] * 2
+                                     + [32] * 3 + [64] * 4 + [96] * 3 + [160] * 3]
+                channels_group = [int(x * multiplier) for x in [16] + [24] * 2 + [32] * 3
+                                  + [64] * 4 + [96] * 3 + [160] * 3 + [320]]
+                ts = [1] + [6] * 16
+                strides = [1, 2] * 2 + [1] * 2 + [2] + [1] * 3 + [1] * 3 + [2] + [1] * 3
+
+                for in_c, c, t, s in zip(in_channels_group, channels_group, ts, strides):
+                    self.features.add(BottleNeck(in_channels=in_c, channels=c, t=t, stride=s))
+
+                last_channels = int(1280 * multiplier) if multiplier > 1.0 else 1280
+                _add_conv(self.features, last_channels)
+
+                self.features.add(nn.GlobalAvgPool2D())
+
+            self.output = nn.HybridSequential(prefix='output_')
+            self.output.add(
```

Comment:
`with self.output.name_scope():`
and manual prefix can be skipped.

Reply:
A name_scope is indeed necessary.